### PR TITLE
AIOPS-2968: Added Artifacts Poetry Plugin

### PR DESCRIPTION
**Pull request by jacksondelametter**

### CODE REVIEW

**Review by d-ryan-ashcraft — COMMENTED** [review]
S (Substantive): Probably want to ignore the `dist` directory.

File: tests/features/steps/artifacts-poetry-plugin-steps.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+def the_following_artifacts(context):
+    context.packages = {}
+    for row in context.table:
+        context.packages[row["name"]] = Package(
+            name=row["package"], pretty_version=row["version"]
+        )
+
+
+@given("a Python project with dependencies on artifacts {keys}")
+def step_impl(context, keys):
```
Comment (aaron-gary) [inline tests/features/steps/artifacts-poetry-plugin-steps.py:39]: Doesn't seem that keys are used in this method.  I can see that the packages are being used for A & B and they are in the context.packages dict, but might want to be more explicit in loading packages A and B.  i.e. 

for key in keys:
    package = context.packages[key]

File: tests/features/steps/artifacts-poetry-plugin-steps.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    response_package_set = set()
+    for package in context.response_packages:
+        response_package_set.add((package.name, package.pretty_version))
+    split_keys = keys.split(",")
+    compare_package_set = set()
+    for key in split_keys:
+        if key in context.packages:
+            package = context.packages[key]
+            compare_package_set.add((package.name, package.pretty_version))
+    assert compare_package_set.issubset(response_package_set) is True
```
Comment (aaron-gary) [inline tests/features/steps/artifacts-poetry-plugin-steps.py]: minor:  you could drop the "is True".  If it isn't True the assert will fail anyways.

**Review by aaron-gary — APPROVED** [review]

File: tests/features/steps/artifacts-poetry-plugin-steps.py (hunk 1 above)
Comment (jacksondelametter) [reply tests/features/steps/artifacts-poetry-plugin-steps.py:39]: Fixed.

File: tests/features/steps/artifacts-poetry-plugin-steps.py (hunk 2 above)
Comment (jacksondelametter) [reply tests/features/steps/artifacts-poetry-plugin-steps.py]: Fixed.

**Pull request merged**